File: C#/Basic Auth/BasicAuth.cs
using System;
using System.IO;
using System.Net;

public class BasicAuth
{
    static public void Main()
    {
        string baseUrl = "{base url}.jamacloud.com/rest/latest/";

        // Username and password should be stored according
        // to your organization's security policies
        string username = "API_User";
        string password = "********";

        string resource = "projects";

        HttpWebRequest request = (HttpWebRequest)WebRequest.Create(baseUrl + resource);
        request.Method = WebRequestMethods.Http.Get;

        string credentials = System.Convert.ToBase64String(System.Text.Encoding.UTF8.GetBytes(username + ":" + password));
        request.Headers["Authorization"] = "Basic " + credentials;

        HttpWebResponse response = (HttpWebResponse)request.GetResponse();

        Stream stream = response.GetResponseStream();
        StreamReader streamReader = new StreamReader(stream);
        string s = streamReader.ReadToEnd();

        Console.WriteLine(s);
    }
}

File: Python/Basic Auth/basic_auth.py
import urllib2
import base64

base_url = "{base url}/rest/latest/"

# Username and password should be stored according
# to your organization's security policies
username = "API_User"
password = "********"

resource = "projects"

request = urllib2.Request(base_url + resource)
base64string = base64.encodestring('%s:%s' % (username, password)).replace('\n', '')
request.add_header("Authorization", "Basic %s" % base64string)

response = urllib2.urlopen(request);

print response.read();
